### ISSUE TITLE
No parameter translator functions specified for kernel 'sos' or language 'sos'

### ISSUE THREAD

**Issue by sonnylaskar** [vatlab/sos-papermill#3]
Currently papermill has no parameter translation function.
I have the below notebook:
![image](https://user-images.githubusercontent.com/7631261/65041205-d0db2500-d973-11e9-9141-15f931a885d4.png)

On running papermill, I see the below error:
```
$ papermill --engine sos U1.ipynb U1_o.ipynb -y '{"x": 11}'
INFO: Input Notebook:  U1.ipynb
INFO: Output Notebook: U1_o.ipynb
Traceback (most recent call last):
  File "/home/ubuntu/.local/bin/papermill", line 11, in <module>
    sys.exit(papermill())
  File "/home/ubuntu/.local/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/papermill/cli.py", line 254, in papermill
    cwd=cwd,
  File "/home/ubuntu/.local/lib/python3.6/site-packages/papermill/execute.py", line 81, in execute_notebook
    nb = parameterize_notebook(nb, parameters, report_mode)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/papermill/parameterize.py", line 78, in parameterize_notebook
    param_content = translate_parameters(kernel_name, language, parameters)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/papermill/translators.py", line 278, in translate_parameters
    return papermill_translators.find_translator(kernel_name, language).codify(parameters)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/papermill/translators.py", line 26, in find_translator
    kernel_name, language
papermill.exceptions.PapermillException: No parameter translator functions specified for kernel 'sos' or language 'sos

```

Can you share some directions on how to get this closed?
[Translators](https://github.com/nteract/papermill/blob/master/papermill/translators.py) is the place where I can see the translators.

**Comment by BoPeng** [comment]
Please check out the master branch of `sos-papermill` and `sos-notebook` and test if this works now.

**Comment by BoPeng** [comment]
A problem here is that the parameter cell will always be `SoS` cell (at least this is no way to specify kernel from command line), so your notebook will not work.

![image](https://user-images.githubusercontent.com/9889312/65050970-e168ac80-d92d-11e9-9f9a-225488005bfb.png)

It is possible to let the "parameters" cell use the kernel of its previous cell (in this case `Python3`), but then a different translator will have to be used. The key here is that if the translater knows other meta information of the cell.

**Comment by BoPeng** [comment]
According to [this line of code](https://github.com/nteract/papermill/blob/master/papermill/parameterize.py#L78), papermill only use global kernel name (`sos`) for the translation of parameters, so there is no way to enable cell kernel-specific parameter translater.

**Comment by BoPeng** [comment]
Note that we could try to be clever, and add `%put param --to kernel` in the `injected-parameter` cell. However, as shown in the following example, papermill collects all parameters and inject a single cell, so the `%put` idea will fail if there are multiple `parameters` cells with different kernels.

```
papermill --engine sos sos1.ipynb sos2.ipynb -y '{"x": 11, "cmd": "echo"}' 
```

![image](https://user-images.githubusercontent.com/9889312/65053900-51793180-d932-11e9-9834-be5a73c01f98.png)

**Comment by sonnylaskar** [comment]
Hi,

Thanks for the super quick turnaround.
It just works!
We will test this in detail and revert back with any further issues if seen.

I think the suggestion to have **parameters** in **SoS** kernel is a fair one and fits much use cases. And then by using the magic of **%magics** of SoS kernel, we can achieve the flow.

**Comment by BoPeng** [comment]
I realized that at least for now papermill only supports one parameters cell (nteract/papermill#328). That is to say we do not have to worry about multiple parameters defined in multiple kernels, so a simple `%put` magic should do the trick. I have submitted a patch to support non-SoS parameters.

 ```
 papermill --engine sos with_non_sos_param.ipynb sos1.ipynb -y '{"x": "Good Bye"}' 
```

![image](https://user-images.githubusercontent.com/9889312/65065846-da02cc80-d948-11e9-8917-c449c953f169.png)
